### ISSUE TITLE
`.*` not defined for sparse matrices

### ISSUE THREAD

**Issue by mauro3** [JuliaLang/julia#11475]
```
julia> sp = sprand(3,3,0.1)
3x3 sparse matrix with 1 Float64 entries:
        [2, 2]  =  0.885289

julia> [1,1,1].* sp
ERROR: DimensionMismatch("")
 in .* at sparse/sparsematrix.jl:553
 in .* at sparse/sparsematrix.jl:655
```

note that broadcasting itself is also broken for sparse matrices: #11474

**Comment by HalcyonChimera** [comment]
I can't seem to reproduce this.

**Comment by JeffBezanson** [comment]
Maybe this was fixed for 0.4?

**Comment by mauro3** [comment]
Yes, it is 0.3.  And it works as expected.  Oddly though `broadcast` makes it full: see #11475

**Comment by tkelman** [comment]
I think this was fixed by #10106 which is a bit of a large change to consider backporting. Okay to close?

**Comment by mauro3** [comment]
Yes, the `broadcast!` workaround of yours in #11474 is good enough.
